### PR TITLE
[TEST] Remove prefixes / suffices in test suites that are not necessary

### PR DESCRIPTION
**Pull request by Irallia**
Please double check that I have not deleted too much. (I've also renamed a few variables, but I'm not sure if they were meant aswell.)

I don't know if there are "other common prefixes".

I left the prefix `global_` if there was also a `local_` prefix.

### CODE REVIEW

File: test/unit/alphabet/custom_alphabet2_test.cpp [hunk 1]
```diff
@@ -26,7 +26,7 @@ class my_alph
     constexpr my_alph(my_alph const &) = default;
     constexpr my_alph & operator=(my_alph const &) = default;
 
-    constexpr my_alph(bool _b) : b{_b} {}
+    constexpr my_alph(bool b) : b{b} {}
```
[the remark this answers is not in the document]
Comment (eseiler) [reply test/unit/alphabet/custom_alphabet2_test.cpp:29]: It's legal for the member initialiser `b{b}`
In the body you would have to use `this->b` to access the class member.

There may be a better name than `b` though :)

File: test/performance/range/views/view_kmer_hash_benchmark.cpp [hunk 2]
```diff
@@ -26,14 +26,14 @@ inline benchmark::Counter bp_per_second(size_t const basepairs)
...
 
     for (size_t i{0}; i < k - 1; ++i)
-        shape_.push_back((i + 1) % 2);
+        shape.push_back((i + 1) % 2);
 
-    shape_.push_back(1u);
-    shape_.push_back(0u);
-    return shape_;
+    shape.push_back(1u);
+    shape.push_back(0u);
```
Comment (Irallia) [inline test/performance/range/views/view_kmer_hash_benchmark.cpp:35]:
```suggestion
    shape.push_back(1u);
    FOO;
```

File: test/unit/alignment/aligned_sequence_test_template.hpp [hunk 3]
```diff
@@ -20,20 +20,20 @@
 using seqan3::operator""_dna4;
 
 template <typename T>
```
Comment (eseiler) [inline test/unit/alignment/aligned_sequence_test_template.hpp:22]:
```suggestion
template <typename t>
```

File: test/unit/range/decorator/gap_decorator_test.cpp [hunk 4]
```diff
@@ -37,26 +37,26 @@ using test_types = ::testing::Types<decorator_t, decorator_t2>;
 // ---------------------------------------------------------------------------------------------------------------------
 
 template <typename inner_type_>
```
Comment (eseiler) [inline test/unit/range/decorator/gap_decorator_test.cpp:39]:
```suggestion
template <typename inner_type>
```

File: test/unit/range/decorator/gap_decorator_test.cpp [hunk 5]
```diff
@@ -37,26 +37,26 @@ using test_types = ::testing::Types<decorator_t, decorator_t2>;
...
-class aligned_sequence_<seqan3::gap_decorator<inner_type_>> : public ::testing::Test
+class aligned_sequence<seqan3::gap_decorator<inner_type_>> : public ::testing::Test
```
Comment (eseiler) [inline test/unit/range/decorator/gap_decorator_test.cpp:40]:
```suggestion
class aligned_sequence<seqan3::gap_decorator<inner_type>> : public ::testing::Test
```

File: test/unit/core/algorithm/detail/execution_handler_template.hpp [hunk 6]
```diff
@@ -77,10 +77,10 @@ TYPED_TEST_P(execution_handler, execute_as_indexed_sequence_pairs)
          it += chunk_size, pos += chunk_size)
     {
         std::ranges::subrange<range_iterator_t, range_iterator_t> chunk{it, std::next(it, chunk_size)};
-        exec_handler.execute(simulate_alignment_with_range, std::move(chunk), [_pos = pos, &buffer] (auto && res) mutable
+        exec_handler.execute(simulate_alignment_with_range, std::move(chunk), [pos = pos, &buffer] (auto && res) mutable
```
Comment (eseiler) [inline test/unit/core/algorithm/detail/execution_handler_template.hpp:80]:
```suggestion
        exec_handler.execute(simulate_alignment_with_range, std::move(chunk), [pos, &buffer] (auto && res) mutable
```

File: test/unit/std/concept/auxiliary_iterator.hpp [hunk 7]
```diff
@@ -84,7 +84,7 @@ struct test_sized_sentinel : public test_sentinel<input_or_output_iter_value_t<i
 {
     using difference_type = typename iterator_type::difference_type;
 
-    iterator_type _pos;
+    iterator_type pos;
```
Comment (eseiler) [inline test/unit/std/concept/auxiliary_iterator.hpp:87]: Can you also please sort the includes of this file while you are touching it?

File: test/unit/alignment/aligned_sequence_test_template.hpp [hunk 8]
```diff
@@ -7,7 +7,7 @@
 
 #include <gtest/gtest.h>
 
-#include <seqan3/std/iterator>
+#include <iterator>
```
Comment (eseiler) [inline test/unit/alignment/aligned_sequence_test_template.hpp:10]:
```suggestion
#include <seqan3/std/iterator>
```

File: test/unit/core/algorithm/detail/execution_handler_template.hpp [hunk 9]
```diff
@@ -5,21 +5,21 @@
 // shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
 // -----------------------------------------------------------------------------------------------------
 
+#include <seqan3/std/algorithm>
+#include <iterator>
```
Comment (eseiler) [inline test/unit/core/algorithm/detail/execution_handler_template.hpp:9]:
```suggestion
#include <seqan3/std/iterator>
```

File: test/unit/io/record_test.cpp [hunk 10]
```diff
@@ -5,16 +5,16 @@
 // shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
 // -----------------------------------------------------------------------------------------------------
 
+#include <algorithm>
```
Comment (eseiler) [inline test/unit/io/record_test.cpp:8]:
```suggestion
#include <seqan3/std/algorithm>
```

File: test/unit/std/concept/auxiliary_iterator.hpp [hunk 11]
```diff
@@ -7,14 +7,13 @@
 
 #pragma once
 
+#include <forward_list>
 #include <iterator>
```
Comment (eseiler) [inline test/unit/std/concept/auxiliary_iterator.hpp:11]:
```suggestion
#include <seqan3/std/iterator>
```

**Pull request merged**